newlib/libc/tinystdio/mktemp.c

```diff
@@ -98,8 +98,10 @@ _gettemp (char *path,
           int *doopen,
           int flags)
 {
-  char *start, *trv;
+  char *trv;
   char *end;
+  static char inc = 0;
```

Comment:
Hrm. How do we handle multiple threads calling into this function? Do we use per-thread variables or do we need some locking here?

Reply:
I think we can just use `__LIBC_LOCK()` and `__LIBC_UNLOCK()` to handle multiple threads calling into this function.
I have added a new commit. 
@keith-packard could you please review this change?